VersionControlUI/src/items/VDiffComparisonPair.cpp

```diff
@@ -71,25 +190,46 @@ void VDiffComparisonPair::initializeForms()
 			->setColumnStretchFactor(0, 1)
 			->setColumnStretchFactor(1, 1)
 			->put(0, 0, oldVersionObjectPath)
-			->put(1, 0, newVersionObjectPath)
+			->put(1, 0, newVersionObjectPath);
+
+	auto infoWithComponentTypeGrid = (new Visualization::GridLayoutFormElement{})
+			->setHorizontalSpacing(50)
+			->setLeftMargin(10)
+			->setRightMargin(10)
+			->setHorizontalAlignment(Visualization::LayoutStyle::Alignment::Center)
+			->setNoBoundaryCursors([](Item*){return true;})->setNoInnerCursors([](Item*){return true;})
+			->setColumnStretchFactor(0, 1)
+			->setRowStretchFactor(0, 1)
+			->put(0, 0, infoGrid->clone())
 			->put(0, 1, componentType);
-	auto container = (new Visualization::GridLayoutFormElement{})
+
+	diffGrid = (new Visualization::GridLayoutFormElement{})
+				->setHorizontalSpacing(50)
+				->setLeftMargin(10)
+				->setRightMargin(10)
+				->setHorizontalAlignment(Visualization::LayoutStyle::Alignment::Center)
+				->setNoBoundaryCursors([](Item*){return true;})->setNoInnerCursors([](Item*){return true;})
+				->setColumnStretchFactor(0, 1)
+				->setColumnStretchFactor(1, 1)
+				->setRowStretchFactor(0, 1)
+				->setColumnHorizontalAlignment(0, Visualization::LayoutStyle::Alignment::Right)
+				->setColumnHorizontalAlignment(1, Visualization::LayoutStyle::Alignment::Left)
+				->put(0, 0, oldVersion)
+				->put(1, 0, newVersion);
+
+	container = (new Visualization::GridLayoutFormElement{})
 				->setHorizontalSpacing(30)
 				->setVerticalSpacing(15)
 				->setHorizontalAlignment(Visualization::LayoutStyle::Alignment::Center)
 				->setNoBoundaryCursors([](Item*){return true;})->setNoInnerCursors([](Item*){return true;})
 				->setLeftMargin(10)
 				->setRightMargin(10)
 				->setColumnStretchFactor(0, 1)
-				->setColumnStretchFactor(1, 1)
-				->setColumnHorizontalAlignment(0, Visualization::LayoutStyle::Alignment::Right)
-				->setColumnHorizontalAlignment(1, Visualization::LayoutStyle::Alignment::Left)
-				->put(0, 0, infoGrid)
-				->setCellSpanning(2, 1)
-				->put(0, 1, oldVersion)
-				->put(1, 1, newVersion);
+				->setColumnHorizontalAlignment(0, Visualization::LayoutStyle::Alignment::Center)
+				->put(0, 0, infoWithComponentTypeGrid->clone())
+				->put(0, 1, diffGrid->clone());
 
-	addForm(container);
+	addForm(container->clone());
```

Comment:
So here, you have a memory leak. You have the container variable, which you always cloned, and now it's an unused pointer left at the end of the method and will not be deleted. Simply don't clone it once. E.g. here. Then the declarative item infrastructure will take ownership of it.